ckanext/saml2/plugin.py

```diff
@@ -339,3 +392,32 @@ def slo(self):
     def staff_login(self):
         """Default login page for staff members."""
         return self.login()
+
+    def manage_permissions(self):
+        """Admin page."""
+        context = {'model': model,
+                   'user': p.toolkit.c.user,
+                   'auth_user_obj': p.toolkit.c.userobj}
+        try:
+            logic.check_access('sysadmin', context, {})
+        except logic.NotAuthorized:
+            base.abort(401, p.toolkit._('Need to be system administrator to administer'))
```

Comment:
This should return a 403 Forbidden unless the user is unauthenticated, for which case a 401 would be correct.